config/routes.rb

```diff
@@ -36,4 +36,16 @@
   get "/veterinary_offices/:veterinary_office_id/veterinarians", to: "veterinary_offices#veterinarians"
   get "/veterinary_offices/:veterinary_office_id/veterinarians/new", to: "veterinarians#new"
   post "/veterinary_offices/:veterinary_office_id/veterinarians", to: "veterinarians#create"
+
+  get "/applications/new", to: "applications#new"
+  post "/applications", to: "applications#create" 
+  get "/applications/:id", to: "applications#show"
+  patch "/applications/:id/update", to: "applications#update"
+
+  post "/pet_apps", to: "pet_apps#create"
+  patch "/pet_apps/:id", to: "pet_apps#update"
```

Comment:
For these join routes, I would suggest something like this:

```ruby
post "applications/:id/pets", to: "pet_apps#create"
patch "applications/:application_id/pets/:id", to: "pet_apps#update"
```

This is a more conventional approach, although what you have is not exactly, non-restful.